samples/UWPTest/publishAll.ps1

```diff
@@ -1,8 +1,61 @@
+$nugetUrl = "https://dist.nuget.org/win-x86-commandline/v3.5.0-beta/NuGet.exe"
+$symchkPath = "C:\Program Files (x86)\Windows Kits\10\Debuggers\x86"
+$msBuildBinFolderPath = "C:\Program Files (x86)\MSBuild\14.0\Bin"
 
-$targetArchs = @("x86", "x64", "arm");
+# Ensure clean state.
+$currentWorkingDirectory = (Get-Item -Path ".\" -Verbose).FullName
+Remove-Item -Path "$currentWorkingDirectory\packages" -Recurse -Force -ErrorAction SilentlyContinue
+Remove-Item -Path "$currentWorkingDirectory\RestoredOutput" -Recurse -Force -ErrorAction SilentlyContinue
+Remove-Item -Path "$currentWorkingDirectory\NuGet.exe" -Force -ErrorAction SilentlyContinue
+Remove-Item -Path "$currentWorkingDirectory\symchk.exe" -Force -ErrorAction SilentlyContinue
+Remove-Item -Path "$currentWorkingDirectory\symbolcheck.dll" -Force -ErrorAction SilentlyContinue
+Remove-Item -Path "$currentWorkingDirectory\Bin\" -Recurse -Force -ErrorAction SilentlyContinue
+Remove-Item -Path "$currentWorkingDirectory\BuildLog.txt" -Force -ErrorAction SilentlyContinue
+
+# Get NuGet
+$nugetLocalPath = Join-Path $currentWorkingDirectory "NuGet.exe"
+$webclient = new-object System.Net.WebClient
+do
+{
+    try
+    {
+		$webclient.DownloadFile($nugetUrl, $nugetLocalPath)
+        break;
+    }
+    catch 
+    {
+        Write-Warning "Error occurred while downloading NuGet.  Exception message follows $_"
+        Start-Sleep -Seconds 2
+    }
+    
+} while ($true)
+
+
+# Get SymbolCheck
+Copy-Item -Path $symchkPath\symchk.exe -Destination "$currentWorkingDirectory\symchk.exe" -Force -ErrorAction Continue
+Copy-Item -Path "$symchkPath\SymbolCheck.dll" -Destination "$currentWorkingDirectory\SymbolCheck.dll" -Force -ErrorAction Continue
+
+# Get MSBuild
+Copy-Item -Path $msBuildBinFolderPath -Destination "$currentWorkingDirectory\Bin" -Recurse -Force -ErrorAction Continue
+
+$targetArchs = @("x86", "x64", "arm")
 
 foreach ($targetArch in $targetArchs)
 {
-    msbuild UWPTest.proj /p:TargetArch=$targetArch
-    msbuild UWPTest.proj /p:TargetArch=$targetArch /p:IsNetNative=true
-}
\ No newline at end of file
+    $msBuildPath = "$currentWorkingDirectory\Bin\MSBuild.exe"
+    $msBuildArgs = "UWPTest.proj /p:TargetArch=$targetArch /verbosity:diag /fileLoggerParameters:LogFile=.\BuildLog.txt;Append;Verbosity=diagnostic;Encoding=UTF-8"
+	
+    $p = Start-Process -FilePath $msBuildPath -ArgumentList $msBuildArgs -PassThru -Verbose -Wait -ErrorAction Continue
```

Comment:
` & $msbuildPath $msBuildArgs ` that would work too.  Start-Process gives a better control.  For example, say the path to MSBuild is invalid then the process exit code will not be 1, and the displayed exception message will help investigate.